File: go/binary_search/binary_search_exists.go
package binary_search

// existsInt should return true if val exists in a sorted array arr, or false if not found
func existsInt(arr []int, val int) bool {
	l, r := 0, len(arr)

	for l < r {
		m := (l + r) / 2

		if arr[m] < val {
			l = m + 1
		} else {
			r = m
		}
	}

	return l < len(arr) && arr[l] == val
}

File: java/src/binary_search/BinarySearchExists.java
package binary_search;

public class BinarySearchExists {
	// existsInt should return true if val exists in a sorted array arr, or false if not found
	public static boolean existsInt(int[] arr, int val) {
		int l = 0, r = arr.length;

		while (l < r) {
			int m = (l + r)/2;
			if (arr[m] >= val) {
				r = m;
			} else {
				l = m + 1;
			}
		}

		return l < arr.length && arr[l] == val;
	}
}
